cosmoz-treenode-navigator.js

```diff
@@ -162,108 +163,45 @@
 		/**
 		 * Returns a node array with the children of the given path.
 		 */
-		_renderLevel: function (pl, nodes) {
-			var pathSegment = nodes,
-				pathArray,
-				level = [];
-			if (!pl) {
-				// Return the formatted root nodes.
-				level = Object.keys(nodes).map(function (key) {
-					var node = nodes[key];
-					return {
-						id: key,
-						name: node[this.comparisonProperty],
-						path: node.pathLocator,
-						children: node[this.childProperty]
-					};
-				}, this);
-				return this._sortItOut(level);
+		_renderLevel: function (pathLocator, tree) {
+			if (!tree) {
+				return;
 			}
-
-			pathArray = pl.split(this.separatorSign);
-			pathArray.forEach(function (pathKey, i, arr) {
-				var node = pathSegment[pathKey],
-					children = node[this.childProperty];
-				if (i === arr.length - 1) {
-					level = Object.keys(children).map(function (childKey) {
-						var child = children[childKey];
-						return {
-							id: child.key,
-							name: child[this.comparisonProperty],
-							path: child.pathLocator,
-							children: child[this.childProperty]
-						};
-					}, this);
-				} else {
-					pathSegment = children;
-				}
-			}, this);
-
-			return this._sortItOut(level);
+			var node = tree.getNodeByPathLocator(pathLocator),
+				children = tree.getChildren(node),
+				level = children || _objectValues(node);
```

Comment:
Why do we need `_objectValues()` on the node here?
Seems like it would render a weird result?